idea/src/org/jetbrains/kotlin/idea/intentions/ValToObjectIntention.kt

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright 2010-2016 JetBrains s.r.o.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.jetbrains.kotlin.idea.intentions
+
+import com.intellij.openapi.editor.Editor
+import com.intellij.refactoring.rename.RenameProcessor
+import org.jetbrains.kotlin.idea.core.replaced
+import org.jetbrains.kotlin.psi.*
+import org.jetbrains.kotlin.psi.psiUtil.endOffset
+import org.jetbrains.kotlin.psi.psiUtil.visibilityModifier
+
+class ValToObjectIntention : SelfTargetingIntention<KtProperty>(KtProperty::class.java, "Convert to object declaration") {
+
+    override fun isApplicableTo(element: KtProperty, caretOffset: Int): Boolean {
+        if (element.isVar) return false
+        if (!element.isTopLevel) return false
+        if (element.initializer !is KtObjectLiteralExpression) return false
+        if (element.getter != null) return false
+
+        return true
+    }
+
+    override fun applyTo(element: KtProperty, editor: Editor?) {
+        val modifier = element.visibilityModifier()
+        val name = element.name ?: return
+        val o = element.initializer as? KtObjectLiteralExpression ?: return
+        val od = o.objectDeclaration
+        val superTypeList = od.getSuperTypeList() ?: return
```

Comment:
I think you should check the supertype list in `isApplicableTo` as well, so that you don't end up with an intention which is enabled and does nothing.